float64[] right_avoidance